### PR TITLE
Adding inheritablePreRun, inheritablePostRun, preRun, postRun

### PR DESCRIPTION
**Pull request by oarrabi**
Adding inheritablePreRun, inheritablePostRun, preRun, postRun
Removing unneeded enum from error
Adding some documentation on CommandType
@goyox86

### CODE REVIEW

File: Sources/Guaka/CommandType/CommandType+Utilities.swift [hunk 1]
```diff
@@ -0,0 +1,90 @@
+//
```
Comment (oarrabi) [inline Sources/Guaka/CommandType/CommandType+Utilities.swift]: This is just splitting CommandType to more files, it should not have actual code change

File: Sources/Guaka/Types/Errors.swift [hunk 2]
```diff
@@ -27,10 +26,7 @@ extension CommandErrors {
   var error: String {
     switch self {
     case let .flagNotFound(flag):
```
Comment (oarrabi) [inline Sources/Guaka/Types/Errors.swift]: These errors were not used

File: Sources/Guaka/CommandType/CommandType+Execution.swift [hunk 3]
```diff
@@ -11,9 +11,28 @@ extension CommandType {
   
   public func execute(flags: [String: Flag], args: [String]) {
     printDeprecationMessages(flags: Array(flags.values))
+    
+    if
```
Comment (goyox86) [inline Sources/Guaka/CommandType/CommandType+Execution.swift]: I get the intention of the if let thingy but it's difficult to read xD

Comment (goyox86) [inline Sources/Guaka/CommandType/CommandType+Execution.swift]: At least in this particular case

File: Sources/Guaka/CommandType/CommandType+Execution.swift [hunk 4]
```diff
@@ -27,7 +46,29 @@ extension CommandType {
...
+}
+
+
+extension CommandType {
+  func findAdequateInheritableRun(forPre: Bool) -> Any?  {
+    
+    var cmd: CommandType? = self
+    
+    while true {
+      var toFind: Any?
```
Comment (goyox86) [inline Sources/Guaka/CommandType/CommandType+Execution.swift]: Do we need to use `Any` here? Don't we have something like the "Closure/Function" Type?

Comment (oarrabi) [inline Sources/Guaka/CommandType/CommandType+Execution.swift]: I rethought this and made `inheritablePostRun` same type of `inheritablePreRun`. Now there is no more Any?

File: Sources/Guaka/CommandType/CommandType+Execution.swift (hunk 3 above)
Comment (oarrabi) [inline Sources/Guaka/CommandType/CommandType+Execution.swift]: I added a fixme to track it and refactor it in the future

**Comment by oarrabi** [comment]
I added a fixme, but I need to merge this since I am building on it :)

**Pull request merged**